javaagent/Dockerfile

```diff
@@ -1,4 +1,4 @@
-FROM eclipse-temurin:11-jre
+FROM eclipse-temurin:21-jre
```

Comment:
thanks for catching this!

if you add this to the `javaagent/build.gradle.kts` file, it should work with 17, which would be ideal since that's the minimum version it will work with now that we've bumped the app to Spring Boot 3

```
java {
    toolchain {
        languageVersion.set(JavaLanguageVersion.of(17))
    }
}
```